mail_tracking/models/mail_message.py

```diff
@@ -103,10 +125,66 @@ def _message_read_dict_postprocess(self, messages, message_tree):
         mail_messages = self.browse(mail_message_ids)
         partner_trackings = mail_messages.tracking_status()
         email_cc = mail_messages._get_email_cc()
+        failed_message = self._get_failed_message()
         for message_dict in messages:
             mail_message_id = message_dict.get('id', False)
             if mail_message_id:
-                message_dict['partner_trackings'] = \
-                    partner_trackings[mail_message_id]
-                message_dict['email_cc'] = email_cc[mail_message_id]
+                message_dict.update({
+                    'partner_trackings': partner_trackings[mail_message_id],
+                    'email_cc': email_cc[mail_message_id],
+                    'failed_message': failed_message[mail_message_id],
+                })
         return res
+
+    @api.model
+    def _prepare_dict_failed_message(self, message):
+        failed_trackings = message.mail_tracking_ids.filtered(
+            lambda x: x.state in ['error', 'rejected', 'spam', 'bounced',
+                                  'soft-bounced'])
+        failed_partners = failed_trackings.mapped('partner_id')
+        failed_recipients = [[partner.id, partner.name]
+                             for partner in failed_partners]
+        return {
+            'id': message.id,
+            'date': message.date,
+            'author_id': [message.author_id.id,
+                          message.author_id.display_name],
+            'body': message.body,
+            'subtype_id': self.env.ref(
+                'mail_tracking.mail_message_subtype_failed_retry').id,
+            'failed_recipients': failed_recipients,
+        }
+
+    @api.multi
+    def get_failed_messages(self):
+        failed_messages = []
+        for msg in self:
+            failed_messages.append(self._prepare_dict_failed_message(msg))
+        return failed_messages
+
+    @api.multi
+    def toggle_tracking_status(self):
+        """Toggle message tracking action needed to ignore them in the tracking
+           issues filter"""
+        # a user should always be able to star a message he can read
+        self.check_access_rule('read')
+        self.track_needs_action = not self.track_needs_action
+        return self.track_needs_action
+
+    def _get_failed_message_domain(self):
+        return [
+            ('mail_tracking_ids.state', 'in',
+                ['error', 'rejected', 'spam', 'bounced', 'soft-bounced']),
+            ('track_needs_action', '=', True)
+        ]
+
+    @api.model
+    def get_failed_count(self):
+        return self.env[self._name].search_count(
+            self._get_failed_message_domain())
+
+    @api.model
+    def message_fetch(self, domain, limit=20):
+        if self.env.context.get('filter_failed_message'):
+            domain = self._get_failed_message_domain()
```

Comment:
Shouldn't it be like this?

```suggestion
            domain = self._get_failed_message_domain() + domain
```

Reply:
No, this is another "hack" related with https://github.com/OCA/social/pull/411/files#diff-d84e32af67b478b77fc10bc4225d49f9R147